src/java/org/apache/cassandra/db/compression/CompressionDictionary.java

```diff
@@ -137,6 +137,41 @@ static CompressionDictionary deserialize(DataInput input, @Nullable CompressionD
         return dictionary;
     }
 
+    static CompressionDictionary createFromRowSimple(UntypedResultSet.Row row)
+    {
+        String kindStr = row.getString("kind");
+        long dictId = row.getLong("dict_id");
+
+        try
+        {
+            Kind kind = CompressionDictionary.Kind.valueOf(kindStr);
+            return new CompressionDictionary()
+            {
+                @Override
+                public DictId dictId()
+                {
+                    return new DictId(kind, dictId);
+                }
+
+                @Override
+                public byte[] rawDictionary()
+                {
+                    return null;
```

Comment:
for listing purposes we do not need raw dict returned at all, we do not need to fetch it from db either hence it is all faster etc

Reply:
Is `list` command necessary? User can query the table directly, and by going through CQL, authorization is applied. (JMX route has authN/authZ too, but CQL is generally more available)